### DISCUSSION TITLE
Files with extension .csv uploads as .txt files

### DISCUSSION THREAD

**Discussion by adilupu7** [livewire/livewire#4636] [1. Help]
**Description**
Files with extension .csv uploads as .txt files.

**Steps to reproduce**
Try to upload .csv file. In livewire-tmp folder it will be .txt.

**Context**
Livewire version: 2.10.1
Laravel version: 8.82.0
Browser: Microsoft Edge Version 98.0.1108.43 (Official build) (64-bit)

**Comment by joshhanley** [comment]
@adilupu7 I just did a test locally and csv files upload to my temp directory as `.csv`. So maybe there is an issue with your configuration?

**Reply by twatwak** [reply]
same issue

**Comment by lusattdev** [comment]
This happens because of the mime type of the csv being "text/plain", tmp uploaded files get their extensions by guessing it according to their mime type. Your csv needs to be of mime type "text/csv" or you'll implement the workaround stated here: https://github.com/livewire/livewire/issues/1999#issuecomment-794414767
